github-status/handler.go

```diff
@@ -153,6 +169,118 @@ func reportStatus(status string, desc string, statusContext string, event *sdk.E
 	return nil
 }
 
+func reportCheck(commitStatus *sdk.CommitStatus, event *sdk.Event) error {
+	ctx := context.Background()
+
+	status := commitStatus.Status
+	url := buildPublicStatusURL(commitStatus.Status, commitStatus.Context, event)
+
+	log.Printf("Check: %s, Context: %s, GitHub AppID: %d, Repo: %s, Owner: %s", status, commitStatus.Context, event.InstallationID, event.Repository, event.Owner)
+
+	client := auth.MakeClient(ctx, token)
+
+	now := github.Timestamp{time.Now()}
+
+	logs, err := getLogs(commitStatus, event)
+	if err != nil {
+		return err
+	}
+
+	checks, _, _ := client.Checks.ListCheckRunsForRef(ctx, event.Owner, event.Repository, event.SHA, &github.ListCheckRunsOptions{CheckName: &commitStatus.Context})
+
+	checkRunStatus := getCheckRunStatus(&status)
+	conclusion := getCheckRunConclusion(&status)
+	log.Printf("Check run status: %s", checkRunStatus)
+
+	var apiErr error
+	if *checks.Total == 0 {
+		check := github.CreateCheckRunOptions{
+			StartedAt: &now,
+			Name:      commitStatus.Context,
+			HeadSHA:   event.SHA,
+			Status:    &checkRunStatus,
+			Output: &github.CheckRunOutput{
+				Text:    formatLogs(&logs),
+				Title:   getCheckRunTitle(commitStatus),
+				Summary: &commitStatus.Description,
+			},
+		}
+
+		if checkRunStatus == "completed" {
+			check.Conclusion = &conclusion
+			check.CompletedAt = &now
+		}
+		log.Printf("Creating check run %s", check.Name)
+		_, _, apiErr = client.Checks.CreateCheckRun(ctx, event.Owner, event.Repository, check)
+	} else {
+		check := github.UpdateCheckRunOptions{
+			Name:       *checks.CheckRuns[0].Name,
+			DetailsURL: &url,
+			Output: &github.CheckRunOutput{
+				Text:    formatLogs(&logs),
```

Comment:
This is a very valid point. I’m not sure if it gives better performance. Theres a treadoff exists, pass by value which will be result an allocation on stack memory copy operation, where in case of pointer it will be in the heap and need to be dereferenced . Memory copy is extremely fast these days. For stack it will allocate and deallocate as the function gets called, but in Heap its stays on the memory until its garbage collected. The more values placed in the heap, the more work the garbage collector has to do. Overall I also agree on the point by reference it seems cheaper, and I’m okay to use that. 

Theres a quote from Rob Pike

You should use a pointer when a pointer is appropriate. The reasoning 
involves semantics first and efficiency second. Big things are slower 
than small things. Indirection has a cost. Tradeoffs exist. 

-rob